strongback-log-processing/src/org/frc4931/utils/LogDecoder.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * FRC 4931 (http://www.evilletech.com)
+ *
+ * Open source software. Licensed under the FIRST BSD license file in the
+ * root directory of this project's Git repository.
+ */
+package org.frc4931.utils;
+import java.io.BufferedInputStream;
+import java.io.BufferedWriter;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Utility to process data logs generated by strongback into human readable csv format. The path to the input 
+ * file should be specified as the first command line argument. The path to the output is optional, if it is not
+ * specified, the output will be saved in the current directory, with the same filename as the input, and a .csv
+ * extension. <br>
+ * {@code Usage: LogDecoder <input_path> [output_path]}
+ * 
+ * <h1>Log file format:</h1>
+ * <p>
+ * The first three bytes are the ASCII log ({@code 6C 6F 67}). The next byte is the number of elements that have 
+ * been logged {@code n}. The next {@code n} bytes are the number of bytes in each data point, followed by 
+ * {@code n} repetitions of the length of the name of each data point and the name itself in ASCII bytes.
+ * <p>
+ * Now the data is recorded with respect to the number of bytes in each element. 
+ * When Logger is stopped it will finish writing the line it is on and then the terminator: {@code FF FF FF FF}.
```

Comment:
"finish writing the line it is on" is a bit confusing. I would refer to them as "records" rather than lines, because they certainly don't correspond to a line in the file (as there is no such thing in binary files). Perhaps:

> When the Logger is stopped, it will complete writing the current record and terminate the file with {@code FF FF FF FF}.

BTW, the log processor probably can be made to be pretty tolerant, not really caring about a complete last record or the terminator. Those would only appear if the logger was shutdown properly; if they don't appear, you might not have all of the data, but you'll get as much as you can. That's actually a good thing. (If we were storing more critical information, we'd want to use a number of additional techniques to verify all data were written along the way and no ending data was lost before it could be written to disk. But this use case is more of an example of "get as much data as you can".)